### ISSUE TITLE
`conflicting-argument-forms` example from the docs does not report the error

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3714]
https://docs.astral.sh/ty/reference/rules/#conflicting-argument-forms
```py
from typing import reveal_type
from ty_extensions import is_singleton

if flag:
    f = repr  # Expects a value
else:
    f = is_singleton  # Expects a type form

f(int)  # no error
```

[playground](https://play.ty.dev/45fcc2ad-1371-4325-8f98-ea9b1bef1783)

**Comment by carljm** [comment]
It looks like most `conflicting-argument-forms` diagnostics were removed when we added `TypeForm` support. The only remaining cases of `conflicting-argument-forms` would involve three callables where we still explicitly mark a parameter using the old `ParameterForm::Type`: that's `typing.cast`, `typing.assert_type`, and `TypeAliasType`.

I suspect the best answer here is to migrate those three remaining cases to also use `TypeForm` and then remove `ParameterForm::Type` and `conflicting-argument-forms` rule entirely. But if that's hard, in the meantime we could update the docs to use an example with one of those three callables, so it actually does emit the diagnostic.

**Comment by sharkdp** [comment]
> I suspect the best answer here is to migrate those three remaining cases to also use `TypeForm` and then remove `ParameterForm::Type` and `conflicting-argument-forms` rule entirely.

See also this discussion: https://github.com/astral-sh/ruff/pull/25421#discussion_r3312166778

**Comment by carljm** [comment]
Continuing that discussion here for better visibility (cc @dcreager ):

> I don't think we can get rid of the rule yet, because there can still in theory be overloads where an argument matches with a `TypeForm` parameter in one overload and a value-form parameter in another.

That wouldn't currently cause us to emit `conflicting-argument-forms`, both `TypeForm` annotations and non `TypeForm` annotations go through the same normal annotated-type path, and type context will determine how we treat them. In the particular overloaded case you mention, we'll just do speculative contextual type inference both ways and the argument either will or won't match for each overload, just like any other case of distinct annotations for the same parameter in different overloads. This seems like the right thing to me -- I don't think we'd want to go out of our way to emit `conflicting-argument-forms` in this case.

So it still looks to me like the only blocker to full removal of `conflicting-argument-forms` is converting the three callables I mentioned, and removing `ParameterForm`. (Edit: I've got Codex working on a PR for this.)

**Comment by Andrej730** [comment]
A note for anyone stumbling upon error similar to below - `conflicting-argument-forms` was removed in 0.0.49, so you'll either need to drop this rule from the config or pin `ty<0.0.49`.

```
warning[unknown-rule]: Unknown rule `conflicting-argument-forms`
   --> pyproject.toml:162:1
    |
162 | conflicting-argument-forms = "error"
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
```
